make/test.mk

```diff
@@ -63,30 +63,44 @@ e2e-cleanup: get-test-namespace
 .PHONY: test-olm-integration
 ## Runs the OLM integration tests without coverage
 test-olm-integration: push-operator-image olm-integration-setup
-	$(eval DEPLOYED_NAMESPACE := operators)
 	$(call log-info,"Running OLM integration test: $@")
+ifeq ($(OPENSHIFT_VERSION),3)
+	$(eval DEPLOYED_NAMESPACE := operators)
 	$(Q)oc apply -f https://raw.githubusercontent.com/operator-framework/operator-lifecycle-manager/master/deploy/upstream/quickstart/olm.yaml
+endif
 	$(eval package_yaml := ./manifests/devconsole/devconsole.package.yaml)
 	$(eval devconsole_version := $(shell cat $(package_yaml) | grep "currentCSV"| cut -d "." -f2- | cut -d "v" -f2 | tr -d '[:space:]'))
 	$(Q)docker build -f Dockerfile.registry . -t $(DEVCONSOLE_OPERATOR_REGISTRY_IMAGE):$(devconsole_version)-$(TAG) \
 		--build-arg image=$(DEVCONSOLE_OPERATOR_IMAGE):$(TAG) --build-arg version=$(devconsole_version)
 	@docker login -u $(QUAY_USERNAME) -p $(QUAY_PASSWORD) $(REGISTRY_URI)
 	$(Q)docker push $(DEVCONSOLE_OPERATOR_REGISTRY_IMAGE):$(devconsole_version)-$(TAG)
-
-	$(Q)sed -e "s,REPLACE_IMAGE,$(DEVCONSOLE_OPERATOR_REGISTRY_IMAGE):$(devconsole_version)-$(TAG)," ./test/e2e/catalog_source.yaml | oc apply -f -
-	$(Q)oc apply -f ./test/e2e/subscription.yaml
-
-	$(Q)operator-sdk test local ./test/e2e/ --go-test-flags "-v -parallel=1"
+ifeq ($(OPENSHIFT_VERSION),3)
```

Comment:
Since this test doesn't really work on openshift 3, do we really even need these checks anymore? i think we are all accepting that it won't work on openshift 3 for the most part?

Reply:
I would say, let's keep it for now. Some future optimizations with pushing/pulling the image might make it work.